_posts/2018-XX-XX-running-kafka-on-dedicated-nodes.md

```diff
@@ -0,0 +1,341 @@
+---
+layout: post
+title:  "Running Kafka on dedicated Kubernetes nodes"
+date: 2018-XX-XX
+author: jakub_scholz
+---
+
+When running applications on OpenShift or Kubernetes, different pods get colocated on the same nodes.
+That is a very useful, because it helps to decrease costs by better utilizing the available hardware.
+But it can also cause some issues.
+One of them are for example _noisy neighbors_.
+A situation when multiple applications which are intensively utilizing the same resource - for example network bandwidth or disk I/O - meet on the same node.
+How do you avoid such problems when running Apache Kafka on OpenShift or Kubernetes with Strimzi?
+And how do you make sure your Apache Kafka delivers the best performance?
+
+<!--more-->
+
+Most Apache Kafka users want from their clusters the best possible performance.
+Good performance does not only mean that most messages are delivered with super low latency or super high throughput.
+It also means that the performance of the cluster is constant in time and without any significant spikes.
+_Noisy neighbors_ are just one possible cause of performance issues.
+
+There are several ways how to get the best out of Apache Kafka:
+
+* Make sure Kafka pods are not scheduled on the same node as other performance intensive applications
+* Make sure Kafka pods are scheduled on the nodes with the most suitable hardware
+* Use nodes which are dedicated to Kafka only
+
+Since the 0.5.0 release, Strimzi supports all of these.
+This blog post will show you how to use them.
+
+# Pod Anti-Affinity
+
+[Pod affinity and anti-affinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity) is a Kubernetes feature which allows you to specify constrains for pod scheduling.
+It allows users to specify whether the pod should be scheduled on the same node (affinity) as other pod or on a different node (anti-affinity).
+The pods which should be included into the scheduling constrain are specified using a label selector.
+The scheduling could be either preferred or required.
+The preferred scheduling defines a soft constrain - it will give you only best effort guarantee.
+When there is no way schedule the pod according to the constrain, the pod will be scheduled in a way which doesn't match the constrain.
+The required scheduling is sets a hard constrain.
+If the constrain cannot be met, the pod will not be scheduled.
+
+Strimzi supports pod affinity for Kafka, Zookeeper and Kafka Connect.
+You can use it to specify the pods which should never run on the same node as Kafka pods.
+Affinity can be specified in the Custom Resources (CR) under the `affinity` key.
```

Comment:
I agree on _property_ instead of _key. But I do not think we need to list all of them. It would just break the flow of the post. The examples should make this clear.